airflow/models.py

```diff
@@ -2023,6 +2031,21 @@ def __init__(
         self.email = email
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
+
+        self.notifiers = notifiers or {}
+        if self.email:
+            if 'email' in self.notifiers:
+                raise AirflowException("Cannot have email and notifier.email \
+                    specified at the same time")
+
+            email_on = {}
+            if self.email_on_retry:
+                email_on[State.UP_FOR_RETRY] = ['text_title.tpl', 'html_body.tpl']
+            if self.email_on_failure:
+                email_on[State.FAILED] = ['text_title.tpl', 'html_body.tpl']
+
+            self.notifiers['email'] = EmailNotifier(email, email_on)
```

Comment:
In theory, yes, but I would need to create 3 of them
- One which only triggers on State.Failed
- One which only triggers on State.UP_FOR_RETRY
- One which triggers on both

I'm not sure if that would actually help, as the notifier class probably doesn't really require much memory.